### ISSUE TITLE
SuperLU failure with new linux clang 3.7.0 target

### ISSUE THREAD

**Issue by permcody** [idaholab/moose#6855]
### Description of the enhancement or error report

After Jason update the build boxes yesterday, one of the MOOSE tests started failing that has been working fine for awhile. It appears to be dying in the SuperLU logic of the solve:

```
#0  0x00007fa7b58bdcc9 in __GI_raise (sig=sig@entry=6) at ../nptl/sysdeps/unix/sysv/linux/raise.c:56
#1  0x00007fa7b58c1218 in __GI_abort () at abort.c:118
#2  0x00007fa7b58fa394 in __libc_message (do_abort=do_abort@entry=1, fmt=fmt@entry=0x7fa7b5a08b28 "*** Error in `%s': %s: 0x%s ***\n") at ../sysdeps/posix/libc_fatal.c:175
#3  0x00007fa7b590666e in malloc_printerr (ptr=<optimized out>, str=0x7fa7b5a08ca0 "free(): invalid next size (normal)", action=1) at malloc.c:4996
#4  _int_free (av=<optimized out>, p=<optimized out>, have_lock=0) at malloc.c:3840
#5  0x00007fa7b98ba82e in Destroy_LU () from /opt/moose/petsc/openmpi_petsc-3.6.4/clang-opt-superlu/lib/libpetsc.so.3.6
#6  0x00007fa7b9287f6b in MatDestroy_SuperLU_DIST () from /opt/moose/petsc/openmpi_petsc-3.6.4/clang-opt-superlu/lib/libpetsc.so.3.6
#7  0x00007fa7b929eca4 in MatDestroy () from /opt/moose/petsc/openmpi_petsc-3.6.4/clang-opt-superlu/lib/libpetsc.so.3.6
#8  0x00007fa7b94ee502 in PCReset_LU () from /opt/moose/petsc/openmpi_petsc-3.6.4/clang-opt-superlu/lib/libpetsc.so.3.6
#9  0x00007fa7b94684e9 in PCReset () from /opt/moose/petsc/openmpi_petsc-3.6.4/clang-opt-superlu/lib/libpetsc.so.3.6
#10 0x00007fa7b955d9af in KSPReset () from /opt/moose/petsc/openmpi_petsc-3.6.4/clang-opt-superlu/lib/libpetsc.so.3.6
#11 0x00007fa7b96177e7 in SNESReset () from /opt/moose/petsc/openmpi_petsc-3.6.4/clang-opt-superlu/lib/libpetsc.so.3.6
#12 0x00007fa7b9612373 in SNESDestroy () from /opt/moose/petsc/openmpi_petsc-3.6.4/clang-opt-superlu/lib/libpetsc.so.3.6
#13 0x00007fa7bdaf2302 in libMesh::PetscNonlinearSolver<double>::clear() () from /home/moosetest/civet/client_root_1/moose/scripts/../libmesh/installed/lib/libmesh_opt.so.0
#14 0x00007fa7bdaf2aa4 in libMesh::PetscNonlinearSolver<double>::solve(libMesh::SparseMatrix<double>&, libMesh::NumericVector<double>&, libMesh::NumericVector<double>&, double, unsigned int) () from /home/moosetest/civet/client_root_1/moose/scripts/../libmesh/installed/lib/libmesh_opt.so.0
#15 0x00007fa7bdb3275c in libMesh::NonlinearImplicitSystem::solve() () from /home/moosetest/civet/client_root_1/moose/scripts/../libmesh/installed/lib/libmesh_opt.so.0
#16 0x00007fa7bec8709b in NonlinearSystem::solve() () from /home/moosetest/civet/client_root_1/moose/framework/libmoose-opt.so.0
#17 0x00007fa7bee2d07d in FEProblem::solve() () from /home/moosetest/civet/client_root_1/moose/framework/libmoose-opt.so.0
#18 0x00007fa7bf16234e in TimeStepper::step() () from /home/moosetest/civet/client_root_1/moose/framework/libmoose-opt.so.0
#19 0x00007fa7bf0baff7 in Transient::solveStep(double) () from /home/moosetest/civet/client_root_1/moose/framework/libmoose-opt.so.0
#20 0x00007fa7bf0bad2c in Transient::takeStep(double) () from /home/moosetest/civet/client_root_1/moose/framework/libmoose-opt.so.0
#21 0x00007fa7bf0ba9d9 in Transient::execute() () from /home/moosetest/civet/client_root_1/moose/framework/libmoose-opt.so.0
#22 0x00007fa7bedaab56 in MooseApp::run() () from /home/moosetest/civet/client_root_1/moose/framework/libmoose-opt.so.0
#23 0x00000000004019ac in main ()
```
### Rationale for the enhancement or information for reproducing the error

This is repeatable on any of the build boxes with the new package, when running multiapps/sub_cycling_failure.text_failure_max_procs. I don't believe the test itself has anything to do with it other than it has the right ingredients for triggering the error. It's running SuperLU and it's threaded. We aren't even reaching the error point of this problem.
### Identified impact

(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)

May only be limited to this particular clang package - investigating

**Comment by permcody** [comment]
@fdkong - you have the new PETSc on your system - maybe you can re-enable this test and see if it works now that SuperLU has been patched?

**Comment by permcody** [comment]
SuperLU has been patched and is available through PETSc 3.7.6+
